akka-http-cluster-management/src/main/scala/akka/cluster/http/management/ClusterHttpManagement.scala

```diff
@@ -0,0 +1,236 @@
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), None)
+
+  def apply(cluster: Cluster, pathPrefix: String, https: ConnectionContext) =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), None, Some(https))
+
+  def apply(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, Some(asyncAuthenticator), Some(https))
+
+  def apply(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), Some(https))
+
+  def create(cluster: Cluster): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster)
+
+  def create(cluster: Cluster, pathPrefix: String): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), None, None)
+
+  def create(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String]): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, Some(asyncAuthenticator), None)
+
+  def create(cluster: Cluster, https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, None, Some(https))
+
+  def create(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String]): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), None)
+
+  def create(cluster: Cluster, pathPrefix: String, https: ConnectionContext) =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), None, Some(https))
+
+  def create(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, Some(asyncAuthenticator), Some(https))
+
+  def create(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), Some(https))
+}
+
+/**
+ * Class to instantiate an [[akka.cluster.http.management.ClusterHttpManagement]] to
+ * provide an HTTP management interface for [[akka.cluster.Cluster]].
+ */
+class ClusterHttpManagement(
+  cluster:            Cluster,
+  pathPrefix:         Option[String]                     = None,
+  asyncAuthenticator: Option[AsyncAuthenticator[String]] = None,
+  https:              Option[ConnectionContext]          = None) {
+
+  private val settings = new ClusterHttpManagementSettings(cluster.system.settings.config)
+  private implicit val system = cluster.system
+  private implicit val materializer = ActorMaterializer()
+  import system.dispatcher
+
+  private var bindingFuture: Future[Http.ServerBinding] = _
+
+  def start(): Unit = {
```

Comment:
this should be safe from people calling it accidentally multiple times.
I'd suggest making `bindingFuture` an `AtomicReference` and bail if server already started.
Perhaps a return type that includes info on what address we started would be useful here too.